CustomPlugin.Tests/Environment/ConfigValidatorTest.cs

```diff
@@ -0,0 +1,47 @@
+using CustomPlugin.Environment;
+using NUnit.Framework;
+
+namespace CustomPlugin.Tests.Environment
+{
+    public class ConfigValidatorTest
+    {
+        [Test]
+        public void TestValidConfig()
+        {
+            var config = new MiaEnvConfiguration
+            {
+                USERID_HEADER_KEY = "userid",
+                GROUPS_HEADER_KEY = "usergroups",
+                CLIENTTYPE_HEADER_KEY = "clienttype",
+                BACKOFFICE_HEADER_KEY = "isbackoffice",
+                MICROSERVICE_GATEWAY_SERVICE_NAME = "microservice-gateway",
+                CRUD_PATH = "crud-path"
+            };
+
+            Assert.DoesNotThrow(() => ConfigValidator.ValidateConfig(config));
+        }
+
+        [Test]
+        public void TestInvalidConfigAllPropsNull()
+        {
+            var config = new MiaEnvConfiguration();
+
+            Assert.Throws(typeof(InvalidEnvConfigurationException), () => ConfigValidator.ValidateConfig(config));
+        }
+        
+        [Test]
+        public void TestInvalidConfigOnePropNull()
+        {
+            var config = new MiaEnvConfiguration
+            {
+                USERID_HEADER_KEY = "userid",
+                GROUPS_HEADER_KEY = "usergroups",
+                CLIENTTYPE_HEADER_KEY = "clienttype",
+                BACKOFFICE_HEADER_KEY = "isbackoffice",
+                MICROSERVICE_GATEWAY_SERVICE_NAME = "microservice-gateway",
+            };
+
+            Assert.Throws(typeof(InvalidEnvConfigurationException), () => ConfigValidator.ValidateConfig(config));
+        }
+    }
+}
```

Comment:
Unless it's a common practice in C#, it'd be better to have all files ending with a new line

```suggestion
}

```